src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,1346 @@
+        self.temperatureRegulation.notify(new_value)
+
+    def _updateAge(self, parameter=None, attributeName="Actual"):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads the amount of hours the GIS reservoir has been open for from the Orsay server and saves it in the age VA
+        """
+        if parameter is None:
+            parameter = self._gis.ReservoirLifeTime
+        if parameter is not self._gis.ReservoirLifeTime:
+            raise ValueError("Incorrect parameter passed to _updateAge. Parameter should be "
+                             "datamodel.HybridGIS.ReservoirLifeTime. Parameter passed is %s." % parameter.Name)
+        if not attributeName == "Actual":
+            return
+        logging.debug("GIS reservoir lifetime updated to %f hours." % float(self._gis.ReservoirLifeTime.Actual))
+        self.age._set_value(float(self._gis.ReservoirLifeTime.Actual), force_write=True)
+
+    def _updatePrecursorType(self, parameter=None, attributeName="Actual"):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads the type of precursor gas in the GIS reservoir from the Orsay server and saves it in the precursorType VA
+        """
+        if parameter is None:
+            parameter = self._gis.PrecursorType
+        if parameter is not self._gis.PrecursorType:
+            raise ValueError("Incorrect parameter passed to _updatePrecursorType. Parameter should be "
+                             "datamodel.HybridGIS.PrecursorType. Parameter passed is %s." % parameter.Name)
+        if not attributeName == "Actual":
+            return
+        logging.debug("Precursor type changed to %s." % self._gis.PrecursorType.Actual)
+        self.precursorType._set_value(self._gis.PrecursorType.Actual, force_write=True)
+
+    def _setTemperatureTarget(self, goal):
+        """
+        goal (float): temperature in °C to set as a target temperature
+        return (float): temperature in °C the target temperature is set to
+
+        Sets the target temperature of the GIS reservoir to goal °C
+        """
+        logging.debug("Setting target temperature to %f." % goal)
+        self._temperaturePar.Target = goal
+        return float(self._temperaturePar.Target)
+
+    def _setTemperatureRegulation(self, goal):
+        """
+        goal (int): mode to set the temperature regulation to. Must be 0, 1 or 2
+
+        Turns temperature regulation off (if goal = 0), on (if goal = 1) or on in accelerated mode (if goal = 2)
+        """
+        reg = False
+        rush = False
+        if goal == 2:
+            logging.debug("Setting temperature regulation to accelerated mode.")
+            rush = True
```

Comment:
Is that correct that to activate the RegulationRush, you need to set RegulationOn to False?